calico/felix/test/test_selectors.py

```diff
@@ -61,23 +61,34 @@ def test_matches():
     yield check_no_match, 'a not in {"a"}', {"a": "a"}
     yield check_no_match, 'a in {"a", "b"}', {"a": "c"}
     yield check_no_match, 'has(b)', {"a": "b"}
+    yield check_no_match, '!!has(b)', {"a": "b"}
+    yield check_no_match, '! has(a)', {"a": "b"}
+    yield check_no_match, '!has(a)', {"a": "b"}
+    yield check_no_match, '!!! has(a)', {"a": "b"}
+    yield check_no_match, '!!!has(a)', {"a": "b"}
+    yield check_no_match, '!! ! has(a)', {"a": "b"}
+    yield check_no_match, '! !!has(a)', {"a": "b"}
```

Comment:
We should probably check that it's possible to get a positive match when ! is used.